internal/mode/static/state/graph/backend_refs.go

```diff
@@ -127,16 +144,108 @@ func createBackendRef(
 		return backendRef, &cond
 	}
 
+	backendTLSPolicy, err := findBackendTLSPolicyForService(
+		backendTLSPolicies,
+		ref,
+		sourceNamespace,
+		configMapResolver,
+	)
+	if err != nil {
+		backendRef = BackendRef{
+			Weight: weight,
+			Valid:  false,
+		}
+
+		cond := staticConds.NewRouteBackendRefUnsupportedValue(err.Error())
+		return backendRef, &cond
+	}
+
 	backendRef = BackendRef{
-		Svc:    svc,
-		Port:   port,
-		Valid:  true,
-		Weight: weight,
+		Svc:              svc,
+		BackendTLSPolicy: backendTLSPolicy,
+		Port:             port,
+		Valid:            true,
+		Weight:           weight,
 	}
 
 	return backendRef, nil
 }
 
+func findBackendTLSPolicyForService(
+	backendTLSPolicies map[types.NamespacedName]*gatewayv1alpha2.BackendTLSPolicy,
+	ref gatewayv1.HTTPBackendRef,
+	routeNamespace string,
+	configMapResolver *configMapResolver,
+) (*gatewayv1alpha2.BackendTLSPolicy, error) {
+	var beTLSPolicy *gatewayv1alpha2.BackendTLSPolicy
+	refNs := routeNamespace
+	if ref.Namespace != nil {
+		refNs = string(*ref.Namespace)
+	}
+
+	for _, btp := range backendTLSPolicies {
+		btpNs := btp.Namespace
+		if btp.Spec.TargetRef.Namespace != nil {
+			btpNs = string(*btp.Spec.TargetRef.Namespace)
+		}
+		if btp.Spec.TargetRef.Name == ref.Name && btpNs == refNs {
+			beTLSPolicy = btp
+			break
+		}
+	}
+
+	if beTLSPolicy == nil {
+		return nil, nil
+	}
+
+	err := validateBackendTLSPolicy(beTLSPolicy, configMapResolver)
```

Comment:
I wonder if we need to validate BackendTLS Policy separately, even if it doesn't reference any Service referenced in any HTTPRoute?

This way, the user will see the status of the Policy right away.

Reply:
We shouldn't update the status of the policy if it is not attached to the Gateway, so we need to wait until we see if we have a shared Service before we update this status.